python_runner/runner.py

```diff
@@ -35,6 +35,22 @@ def __init__(
     def set_callback(self, callback):
         self._callback = callback
 
+    def set_source_code(self, source_code, filename):
```

Comment:
This signature is weird. Internally, only one argument is really being used at a time. Is it because of something you want to do externally?

Reply:
Well, it is linked to keeping the source code as an attribute. However, to work properly, the source code needs a filename attached to it to have a proper meaning. Both arguments are also used, so I'm not sure what you mean by one at a time?

Reply:
What I meant was that either it was only really setting `filename` (with source_code="") or only really setting source_code (since filename=self.filename doesn't do much).

Reply:
The source code could be different from "" in the constructor. However, setting the file separately might also make more sense. If you are changing between files in your editor, you probably don't want to run it immediately but still inform the runner of changes. I'll split it into two methods.